lib/membrane_rtmp_plugin/rtmp/source/message_parser.ex

```diff
@@ -175,19 +180,23 @@ defmodule Membrane.RTMP.MessageParser do
   # in this case the message gets divided into
   # a sequence of smaller packets separated by the a header type 3 byte
   # (the first 2 bits has to be 0b11)
-  defp combine_body_chunks(body, chunk_size) do
+  defp combine_body_chunks(body, chunk_size, header) do
     if byte_size(body) <= chunk_size do
       body
     else
-      do_combine_body_chunks(body, chunk_size, [])
+      do_combine_body_chunks(body, chunk_size, [], header)
     end
   end
 
-  defp do_combine_body_chunks(body, chunk_size, acc) do
+  defp do_combine_body_chunks(body, chunk_size, acc, header) do
```

Comment:
We see a refactoring opportunity in here as there is code duplication with the deserialise function. Maybe a reduction till all bytes required by the header have been parsed?

Reply:
I would leave it for now, and if you really want it could be a matter of a separate PR.